bindings/python/tests/test_parse.py

```diff
@@ -0,0 +1,332 @@
+
+    struct_decl = parsed_info["members"][0]
+
+    assert struct_decl["kind"] == "STRUCT_DECL"
+    assert struct_decl["name"] == "AStruct"
+
+
+def test_cxx_base_specifier(tmp_path):
+    file_contents = """
+    struct BaseStruct {};
+    struct DerivedStruct: public BaseStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    child_struct_decl = parsed_info["members"][1]
+    cxx_base_specifier = child_struct_decl["members"][0]
+
+    assert cxx_base_specifier["kind"] == "CXX_BASE_SPECIFIER"
+    assert cxx_base_specifier["access_specifier"] == "PUBLIC"
+    assert cxx_base_specifier["name"] == "struct BaseStruct"
+
+
+def test_cxx_method(tmp_path):
+    file_contents = """
+    struct AStruct {
+        void aMethod() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][0]
+
+    assert cxx_method["kind"] == "CXX_METHOD"
+    assert cxx_method["result_type"] == "void"
+    assert cxx_method["name"] == "aMethod"
+
+
+def test_var_decl(tmp_path):
+    file_contents = "int anInt = 1;"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    var_decl = parsed_info["members"][0]
+
+    assert var_decl["kind"] == "VAR_DECL"
+    assert var_decl["element_type"] == "Int"
+    assert var_decl["name"] == "anInt"
+
+
+def test_type_ref(tmp_path):
+    file_contents = """
+    struct AStruct { 
+        int aClassMember;
+        void aMethod(AStruct& aClassMember) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][1]
+    l_value_ref = cxx_method["members"][0]
+    type_ref = l_value_ref["members"][0]
+
+    assert type_ref["kind"] == "TYPE_REF"
+    assert type_ref["name"] == "struct AStruct"
+
+
+def test_constructor(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+
+    assert constructor["kind"] == "CONSTRUCTOR"
+    assert constructor["access_specifier"] == "PUBLIC"
+    assert constructor["name"] == "AStruct"
+
+
+def test_parm_decl(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct(int aFunctionArgument) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+    parm_decl = constructor["members"][0]
+
+    assert parm_decl["kind"] == "PARM_DECL"
+    assert parm_decl["element_type"] == "Int"
+    assert parm_decl["name"] == "aFunctionArgument"
+
+
+def test_call_expr(tmp_path):
```

Comment:
Maybe have a different example here. This one is good, but too many AStruct to confuse a reader.
```py
struct AStruct {};
struct BStruct { BStruct(): AStruct() {} };
```

Reply:
Something without struct, perhaps?
```cpp
    int aFunction() {
        return 1;
    }
    int anInt = aFunction();
```